students/eunhyekim/users/views.py

```diff
@@ -1,3 +1,42 @@
-from django.shortcuts import render
+import json
+import re
 
-# Create your views here.
+from django.http      import JsonResponse
+from django.views     import View
+
+from users.models     import User
+
+class UsersView(View):
+    def post(self, request):
+        data       = json.loads(request.body)
+        email      = data["email"]
+        password   = data["password"]
+        user       = User()
+        try:
+            try:
+                email_validation    = re.compile('^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$')
+                if not email_validation.match(email):
+                    raise Exception   #({"MESSAGE":"KEY_ERROR"})
+            except Exception:
+                return JsonResponse({"email_MESSAGE":"KEY_ERROR"}, status = 400)
+            
+            try:
+                password_validation = re.compile('(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[!@#$%*^&+=])([a-zA-Z0-9!@#$%*^&+=]{8,})')
+                if not password_validation.match(password):
+                    raise Exception   #({"MESSAGE":"KEY_ERROR"})
+            except Exception:
+                return JsonResponse({"password_MESSAGE":"KEY_ERROR"}, status = 400)
+        
+            else:
+                User.objects.create(
+                    name     = data["name"],
+                    email    = data["email"],
+                    password = data["password"],
+                    contact  = data["contact"],
+                )
+            user.full_clean()
```

Comment:
```suggestion
                    contact  = data["contact"],
                )
            user.full_clean()
```

db에 unique가 있는데  uniuser.full_clean을 넣는 이유가 있을까요????

Reply:
unique = True 만을 사용하였을 경우에는, 직접 지정한 에러 메세지를 보여주는 것이 아니라 실제 에러를 리턴해주고 있어서 만약 그 리턴 된 오류에서 실제 에러 메세지(예를 들어 이 경우에는 IntegrityError)를 except로 지정해서 에러 메세지를 리턴해줄 수 있습니다. user.full_clean의 경우, models.py에서 unique = True 속성을 사용한 데이터(이름, 이메일, 기타 정보들도 만약 사용하였다면)들은 모두 중복 검사를 해서 지정한 에러메세지를 리턴할 수 있습니다.